azure/services/asogroups/groups.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package asogroups
```

Comment:
suggestion for code organization: instead of prefixing every folder name with `aso` it might make sense to reorganize the services folder with two sub-folders: `sdk` and `aso` (we can even add `sdkv2` if needed).

Reply:
I think this will be the only service to have both ASO and SDK versions in the repo at the same time. Once we're ready for more services, I think the way the proposal describes of transitioning each service in-place will be easiest. Do you think it's still important to distinguish that in the folder structure?

Reply:
I think so long as we're able to do a targeted, quick revert (ideally just click the github revert PR button) per service it shouldn't matter. If one or the other makes that easier then I'd vote for that.

Sorry this didn't get hashed out in the proposal PR, probably this is too low level for being declared in the proposal, I should have caught that, if so.

Reply:
@CecileRobertMichon Sounds good. We can always adjust later too if needed.

@jackfrancis I think reverting in this case would be as simple as reverting the relevant commits, but we'll verify that as part of #3546.